webhooks/src/main/java/org/openmbee/sdvc/webhooks/controllers/WebhooksController.java

```diff
@@ -0,0 +1,208 @@
+            }
+
+            if (!webhookExists(json)) {
+                try {
+                    if (!mss.hasProjectPrivilege(auth, json.getProjectId(), Privileges.PROJECT_CREATE_WEBHOOKS.name(), false)) {
+                        Map<String, Object> rejection = new HashMap<>();
+                        rejection.put("message", "No permission to create project under org");
+                        rejection.put("code", 403);
+                        rejection.put("request", json);
+                        rejected.add(rejection);
+                        continue;
+                    }
+
+                    Optional<Project> project = projectRepository.findByProjectId(json.getProjectId());
+                    project.ifPresentOrElse(proj -> {
+                        Webhook newWebhook = new Webhook();
+                        newWebhook.setProject(proj);
+                        newWebhook.setUri(json.getUri());
+                        webhookRepository.save(newWebhook);
+                        response.getWebhooks().add(json);
+                    }, () -> {
+                        Map<String, Object> rejection = new HashMap<>();
+                        rejection.put("message", "No project found");
+                        rejection.put("code", 404);
+                        rejection.put("request", json);
+                        rejected.add(rejection);
+                    });
+
+                } catch (BadRequestException ex) {
+                    Map<String, Object> rejection = new HashMap<>();
+                    rejection.put("message", "Org to put project under is not found");
+                    rejection.put("code", 400);
+                    rejection.put("request", json);
+                    rejected.add(rejection);
+                    continue;
+                }
+            } else {
+                Map<String, Object> rejection = new HashMap<>();
+                rejection.put("message", "No permission to change project");
+                rejection.put("code", 403);
+                rejection.put("request", json);
+                rejected.add(rejection);
+                continue;
+            }
+        }
+        return ResponseEntity.ok(response);
+    }
+
+    @DeleteMapping
+    @Transactional
+    @PreAuthorize("isAuthenticated()")
+    public ResponseEntity<? extends BaseResponse> handleDelete(
+        @RequestBody WebhookRequest webhookRequest,
+        @RequestParam(required = false, defaultValue = "false") boolean hard) {
+
+        // TODO: Determine which webhook to delete somehow. Maybe require webhook id.
+
+        Set<String> projects = new HashSet<>();
+        Set<String> uris = new HashSet<>();
+        for (WebhookJson webhookJson : webhookRequest.getWebhooks()) {
+            projects.add(webhookJson.getProjectId());
+            uris.add(webhookJson.getUri());
+        }
+
+        WebhookResponse response = new WebhookResponse();
+        for (String projectId : projects) {
```

Comment:
the relationship between webhook and project id is lost, this can inadvertently delete the wrong webhook